specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -8523,16 +8523,9 @@
           "description": "The query text. Type: string (or Expression with resultType string)."
         },
         "type": {
-          "x-ms-enum": {
-            "name": "ScriptType",
-            "modelAsString": true
-          },
-          "enum": [
-            "Query",
-            "NonQuery"
-          ],
-          "type": "string",
-          "description": "The type of the query. Type: string."
+          "type": "object",
+          "x-ms-format": "dfe-string",
```

Comment:
>"x-ms-format": "dfe-string", [](http://example.com/codeflow?start=10&length=28)

Can you elaborate on what this means and why are you not directly defining this as ScriptType?

Reply:
ScriptType is enum of string before, but ADF need support expressions so we changed from string to object.

https://learn.microsoft.com/en-us/azure/data-factory/control-flow-expression-language-functions